pkg/build/generator/generator.go

```diff
@@ -0,0 +1,243 @@
+package generator
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// BuildGenerator is a central place responsible for generating new Build objects
+// from BuildConfigs and other Builds.
+type BuildGenerator struct {
+	Client GeneratorClient
+}
+
+type GeneratorClient interface {
+	GetBuildConfig(ctx kapi.Context, name string) (*buildapi.BuildConfig, error)
```

Comment:
couldn't this just embed the BuildConfigGetter, BuildConfigUpdater, etc interfaces, rather than redefining the methods?